File: src/net/eduard/api/command/map/MapSaveCommand.java

package net.eduard.api.command.map;

import net.eduard.api.EduardAPI;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import net.eduard.api.lib.modules.Mine;
import net.eduard.api.lib.manager.CommandManager;

public class MapSaveCommand extends CommandManager {

	public MapSaveCommand() {
		super("save","salvar");
		setUsage("/map save <name>");
		setDescription("Salva o Schematic (Mapa) copiado");
	}

	@Override
	public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
		if (args.length == 1) {
			sendUsage(sender);
		} else {

			if (Mine.onlyPlayer(sender)) {
				Player p = (Player) sender;
				if (!EduardAPI.MAPS_CACHE.containsKey(p)) {
					p.sendMessage("§bEduardAPI §aPrimeiro copie um Mapa:§2 /map copy");
					return true;
				}
				EduardAPI.MAPS.put(args[1].toLowerCase(), EduardAPI.MAPS_CACHE.get(p));
				p.sendMessage("§bEduardAPI §aMapa salvado com sucesso!");
			}
		}
		return true;
	}

}

File: src/main/java/net/eduard/api/command/map/MapSaveCommand.kt
package net.eduard.api.command.map

import net.eduard.api.lib.manager.CommandManager
import net.eduard.api.server.minigame.MinigameSchematic
import net.eduard.api.server.minigame.MinigameSchematic.Companion.isEditing
import org.bukkit.entity.Player
import java.io.File

class MapSaveCommand : CommandManager("save", "salvar") {
    override fun playerCommand(player: Player, args: Array<String>) {
        if (args.isEmpty()) {
            sendUsage(player)
            return
        }

        if (!isEditing(player)) {
            player.sendMessage("§bEduardAPI §aPrimeiro copie um Mapa:§2 /map copy")
            return
        }
        val schema = MinigameSchematic.getSchematic(player)
        schema.name = args[0].toLowerCase()
        schema.register()
        schema.save(File(MinigameSchematic.MAPS_FOLDER, schema.name + ".map"))

        player.sendMessage("§bEduardAPI §aMapa salvado com sucesso!")

    }

    init {
        usage = "/map save <name>"
        description = "Salva o Schematic (Mapa) copiado"
    }
}